### PR TITLE
BornAgain: new recipe

### PR DESCRIPTION
**Pull request by tschoonj**
@conda-forge/help-python-c this is a fun one... Works fine on my Mac, and I guess it will work on Linux as well. Will need to put in some work for the Windows build based on CI.

Since I am linking against some of the boost libraries, I was wondering how to add that dependency at run-time... Should I use `pin_compatible` as well here?
How about the Qt/PyQt dependency? PyQt appears to require Qt 5.9, even though Qt 5.12 is available, so I had to hardcode the build to Qt 5.9...

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by xhochy** [comment]
No need for `pin_compatible` for Boost as it has a `run_exports` section that does the pinning automatically.

File: recipes/bornagain/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+  host:
+    - boost
+    - fftw
+    - gsl
+    - qt 5.9.*
```
Comment (xhochy) [inline recipes/bornagain/meta.yaml]: We have Qt pinned globally at 5.12 https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/4f8f7f03140d73db3a0bf144ea537f466ca23fbf/recipe/conda_build_config.yaml#L568-569 Would that work here too?

File: recipes/bornagain/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: http://apps.jcns.fz-juelich.de/src/BornAgain/BornAgain-{{ version }}.tar.gz
+  sha256: 20e390b683b3036aabf48d266e5e74da46ce9d781e72379dccb3d227ed300d40
+
+build:
+  number: 0
```
Comment (xhochy) [inline recipes/bornagain/meta.yaml]: Please skip Python 2 builds explicitly.

```suggestion
  number: 0
  skip: true  # [py2k]
```

File: recipes/bornagain/bld.bat [hunk 3]
```diff
@@ -0,0 +1,22 @@
+mkdir build
+cd build
+cmake ^
+    -G "NMake Makefiles JOM" ^
```
Comment (xhochy) [inline recipes/bornagain/bld.bat]: You could also try out Ninja but JOM should be fine.
```suggestion
    -G "Ninja" ^
```

File: recipes/bornagain/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    - {{ cdt('libxxf86vm') }}  # [linux]
+    - cmake
+    - ninja
+
+  host:
+    - boost
+    - fftw
+    - gsl
+    - qt 5.9.*
+    - python >=3
```
Comment (marcelotrevisani) [inline recipes/bornagain/meta.yaml]: Please remove the ``>=3`` as you are already skipping python 2 this is not necessary
```suggestion
    - python
```

File: recipes/bornagain/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    - numpy
+    - libtiff
+  run:
+    - boost
+    - fftw
+    - gsl
+    - qt 5.9.*
+    - python >=3
```
Comment (marcelotrevisani) [inline recipes/bornagain/meta.yaml]: ditto
```suggestion
    - python
```

File: recipes/bornagain/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+test:
+  imports:
+    - bornagain
+  commands:
+    - bornagain --help
+
+about:
+  home: https://www.bornagainproject.org
+  license: GPL-3.0
```
Comment (marcelotrevisani) [inline recipes/bornagain/meta.yaml]:
```suggestion
  license: GPL-3.0
  license_family: GPL
```

**Comment by tschoonj** [comment]
How should I deal with these RPATH issues:

```
Traceback (most recent call last):
  File ".ci_support/build_all.py", line 134, in <module>
    build_all(args.recipes_dir, args.arch)
  File ".ci_support/build_all.py", line 68, in build_all
    build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
  File ".ci_support/build_all.py", line 124, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/build.py", line 2343, in build_tree
    notest=notest,
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/build.py", line 1642, in build
    newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/build.py", line 1002, in bundle_conda
    files = post_process_files(metadata, initial_files)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/build.py", line 867, in post_process_files
    post_build(m, new_files, build_python=python)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/post.py", line 1150, in post_build
    check_overlinking(m, files)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/post.py", line 1087, in check_overlinking
    m.config.enable_static)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/post.py", line 1051, in check_overlinking_impl
    raise OverLinkingError(overlinking_errors)
conda_build.exceptions.OverLinkingError: overlinking check failed 
['  ERROR (bornagain,lib/BornAgain-1.16/_libBornAgainGUI.so): $RPATH/_libBornAgainCore.so not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?', '  ERROR (bornagain,lib/BornAgain-1.16/_libBornAgainGUI.so): $RPATH/_libBornAgainFit.so not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?', "  ERROR (bornagain,lib/BornAgain-1.16/_libBornAgainCore.so): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (bornagain,lib/BornAgain-1.16/_libBornAgainCore.so): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", '  ERROR (bornagain,lib/BornAgain-1.16/_libBornAgainCore.so): $RPATH/_libBornAgainFit.so not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?', "  ERROR (bornagain,lib/BornAgain-1.16/exec/BornAgain): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (bornagain,lib/python3.6/site-packages/_libBornAgainCore.so): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (bornagain,lib/python3.6/site-packages/_libBornAgainCore.so): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", '  ERROR (bornagain,lib/python3.6/site-packages/_libBornAgainCore.so): $RPATH/_libBornAgainFit.so not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?', '  ERROR (bornagain,lib/python3.6/site-packages/_libBornAgainGUI.so): $RPATH/_libBornAgainCore.so not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?', '  ERROR (bornagain,lib/python3.6/site-packages/_libBornAgainGUI.so): $RPATH/_libBornAgainFit.so not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?']
```

When doing the build on my MacbookPro, these were warnings, not errors...

**Comment by tschoonj** [comment]
Similar problems on Linux now too...

**Comment by tschoonj** [comment]
Giving up now... the Windows build fails due to some C++11/14 feature not being support by VC2015...

**Pull request closed without merging**